### PR TITLE
currency localisation

### PR DESCRIPTION
**Pull request by nasimafeef**

### CODE REVIEW

File: js/browser.js [hunk 1]
```diff
@@ -62,6 +63,35 @@ var Billing = Class(Emitter, function (supr) {
...
+          res['data'] = {};
+
+          _.each(catalog, function (val) {
+            if(_.contains(products, val['productID'])) {
+              id = val['productID'];
+              res['data'][id] = {};
+
+              // Setting currency attribute as empty string
+              // since amount string contains currency symbol
+              res['data'][id].currency = '';
```
Comment (vivekb91) [inline js/browser.js]: we need to verify this with database

File: js/browser.js [hunk 2]
```diff
@@ -62,6 +63,35 @@ var Billing = Class(Emitter, function (supr) {
...
+          });
+          cb(res);
+        })
+        .catch(function (e){
+          cb(res);
+        });
+    } else {
+      cb(res);
+    }
+  };
```
Comment (rampr) [inline js/browser.js:92]: one way this code might be more readable is by returning early

```
if (!fbinstant.payments_ready) {
  cb(resp);
}

// get localized prices logic
fbinstant.getCatalogAsync()
  .then(...)
// rest of the logic
```

Comment (nasimafeef) [reply js/browser.js:92]: done

File: js/browser.js [hunk 3]
```diff
@@ -62,6 +63,33 @@ var Billing = Class(Emitter, function (supr) {
...
+    fbinstant.getCatalogAsync()
+      .then(function (catalog) {
+        res['data'] = {};
+
+        _.each(catalog, function (val) {
+          if(_.contains(products, val['productID'])) {
+            id = val['productID'];
+            res['data'][id] = {};
+            res['data'][id].currency_code = val['priceCurrencyCode'];
+            res['data'][id].amount = val['price'];
```
Comment (rampr) [inline js/browser.js]: suggestion for the future :)
```
res['data'][id] = {
  currency_code: val['priceCurrencyCode'],
  amount: val['price']
};
```

File: js/browser.js [hunk 4]
```diff
@@ -62,6 +63,33 @@ var Billing = Class(Emitter, function (supr) {
...
+    }
+
+    fbinstant.getCatalogAsync()
+      .then(function (catalog) {
+        res['data'] = {};
+
+        _.each(catalog, function (val) {
+          if(_.contains(products, val['productID'])) {
+            id = val['productID'];
+            res['data'][id] = {};
```
Comment (vivekb91) [inline js/browser.js]: we can call like res.data also

File: js/browser.js (hunk 3 above)
Comment (rampr) [reply js/browser.js]: also `_.map` is a good use case for this.

**Review by rampr — APPROVED** [review]

File: js/browser.js (hunk 4 above)
Comment (rampr) [reply js/browser.js]: yes, that is usually preferred.

File: js/browser.js [hunk 5]
```diff
@@ -62,6 +63,33 @@ var Billing = Class(Emitter, function (supr) {
...
+      cb(res);
+      return;
+    }
+
+    fbinstant.getCatalogAsync()
+      .then(function (catalog) {
+        res['data'] = {};
+
+        _.each(catalog, function (val) {
+          if(_.contains(products, val['productID'])) {
```
Comment (vivekb91) [inline js/browser.js]: here also better to check like val.productId

**Review by vivekb91 — APPROVED** [review]

**Pull request merged**